### PR TITLE
readme: begin documentation

### PR DESCRIPTION
**Pull request by joe-lawrence**
Add some words that begin to describe what this project is about.

### CODE REVIEW

**Comment by joe-lawrence** [comment]
It's easier to see the Markdown rendered in HTML [here](https://github.com/joe-lawrence/pg-table_tests/tree/README-malloc).

File: README.md [hunk 1]
```diff
@@ -1,2 +1,88 @@
 # pg-table_tests
 
+This project consists of a suite of tools to test and verify 5-level page table
+behavior on Linux.
+
+As documented in the [Linux kernel](Documentation/x86/x86_64/5level-paging.txt),
+5-level page tables boosts the virtual address space from 256 TiB
+(2<sup>47</sup>) to 128 PiB (2<sup>57</sup>).
+
```
Comment (joe-lawrence) [inline README.md]: should say "2^47 bytes"

File: README.md [hunk 2]
```diff
@@ -1,2 +1,88 @@
...
+Since Linux splits its virtual address space between user and kernel, this means
+that applications can expect to see an increase from 128 TiB (2<sup>46</sup>) to 64 PiB
+(2<sup>56</sup>).  However, due to backwards compatibility concerns, this increase is only
+achievable by using the `mmap()` call with a specified hint address above the
+previous 47-bit limit.
+
+Here is an aligned dump of `/proc/.../maps` for such process, each of the 1GiB
+"rw-p" anonymous chunks were allocated using mmap() with a desired hint address:
+
```
Comment (joe-lawrence) [inline README.md]: formatting nits, "rw-p" and "mmap()" should be in fixed-width code font

**Comment by joe-lawrence** [comment]
v2 : a few formatting updates

File: README.md [hunk 3]
```diff
@@ -1,2 +1,89 @@
...
 
+This project consists of a suite of tools to test and verify 5-level page table
+behavior on Linux.
+
+As documented in the [Linux kernel](Documentation/x86/x86_64/5level-paging.txt),
+5-level page tables boosts the virtual address space from 256 TiB
+(2<sup>47</sup> bytes) to 128 PiB (2<sup>57</sup> bytes).
+
+Since Linux splits its virtual address space between user and kernel, this
+means that applications can expect to see an increase from 128 TiB
```
Comment (sanskriti-s) [inline README.md]: Since Linux splits its virtual address space between user and kernel, applications can....

File: README.md [hunk 4]
```diff
@@ -1,2 +1,89 @@
...
+(2<sup>46</sup> bytes) to 64 PiB (2<sup>56</sup> bytes).  However, due to
+backwards compatibility concerns, this increase is only achievable by using the
```
Comment (sanskriti-s) [inline README.md]: can only be achieved using

File: README.md [hunk 5]
```diff
@@ -1,2 +1,89 @@
...
+`mmap()` call with a specified hint address above the previous 47-bit limit.
+
+Here is an aligned dump of `/proc/.../maps` for such process, each of the 1GiB
+`rw-p` anonymous chunks were allocated using `mmap()` with a desired hint
+address:
+
```
Comment (sanskriti-s) [inline README.md]: ...for such a process. Each of the.....was allocated

File: README.md [hunk 6]
```diff
@@ -1,2 +1,89 @@
...
+4000000000000-4000000001000        rw-p  00000000  00:00  0         
+8000000000000-8000000001000        rw-p  00000000  00:00  0         
+10000000000000-10000000001000      rw-p  00000000  00:00  0         
+20000000000000-20000000001000      rw-p  00000000  00:00  0         
+40000000000000-40000000001000      rw-p  00000000  00:00  0         
+80000000000000-80000000001000      rw-p  00000000  00:00  0         
+ffffffffff600000-ffffffffff601000  r-xp  00000000  00:00  0         [vsyscall]
+```
+
+Notice that the stack is still located toward the end of 4-level page table
```
Comment (sanskriti-s) [inline README.md]: towards the end

File: README.md [hunk 7]
```diff
@@ -1,2 +1,89 @@
...
+
+This feature may be emulated by QEMU's qemu64 cpu with the following option:
+```
+qemu-system-x86_64 ... -cpu qemu64,+la57
+```
+Note: this option is not compatible with `-enable-kvm`.
+
+### Linux config
+
+The Linux kernel itself must have 5-level page table support enabled:
```
Comment (sanskriti-s) [inline README.md]: kernel must have

**Comment by joe-lawrence** [comment]
v3 : grammar corrections suggested by @sanskriti-s

**Pull request merged**